datafusion/src/physical_plan/expressions/average.rs

```diff
@@ -38,11 +38,18 @@ use super::{format_state_name, sum};
 pub struct Avg {
     name: String,
     expr: Arc<dyn PhysicalExpr>,
+    data_type: DataType,
 }
 
 /// function return type of an average
 pub fn avg_return_type(arg_type: &DataType) -> Result<DataType> {
     match arg_type {
+        DataType::Decimal(precision, scale) => {
+            // the new precision and scale for return type of avg function
```

Comment:
Can you please document document the rationale for the `4` and `38` constants below (or even better pull them into named constants somewhere)?

I also don't understand where the additional `4` came from. I tried to see if it was what postgres did, but when I checked the output schema for `avg(numeric(10,3))` appears to be `numeric` without the precision or scale specified 🤔 


```shell
(arrow_dev) alamb@MacBook-Pro-2:~/Downloads$ psql
psql (14.1)
Type "help" for help.

alamb=# create table test(x decimal(10, 3));
CREATE TABLE
alamb=# insert into test values (1.02);
INSERT 0 1
alamb=# create table test2 as select avg(x) from test;
SELECT 1

alamb=# select table_name, column_name, numeric_precision, numeric_scale, data_type from information_schema.columns where table_name='test2';
 table_name | column_name | numeric_precision | numeric_scale | data_type 
------------+-------------+-------------------+---------------+-----------
 test2      | avg         |                   |               | numeric
(1 row)
```

Reply:
This is the intention for https://github.com/apache/arrow-datafusion/issues/1418.
In the PG, we can create bigger precision for decimal than that in datafusion.
The decimal in datafusion, whose behavior may be different from that in PG.
So we should discuss them.
Now for the promotion precision or scale, I just follow the spark behavior.

Reply:
I think following the spark behavior is reasonable, but it should be documented (aka that the constant `4` came from spark).

Reply:
add comments to the code and issue https://github.com/apache/arrow-datafusion/issues/1461 to track this rule.
We can add the rule to our document later in the follow-up pull request.